src/utils.ts

```diff
@@ -665,12 +673,38 @@ export class RequestersList {
       //console.log('reqHistory after clearing heavy request history', reqHistory.length)
     }
 
-    // unban the ip after 1 hour
-    this.bannedIps = this.bannedIps.filter((record: { ip: string; timestamp: number }) => {
-      if (now - record.timestamp >= 60 * 60 * 1000) return false
-      else return true
+    // unban the ip after 1 hour and ensure synchronization
+    const previousLength = this.bannedIps.length
+    this.bannedIps = this.bannedIps.filter((record) => {
+      if (now - record.timestamp >= 60 * 60 * 1000) {
+        if (verbose) console.log(`Removing IP ${record.ip} from banned list`)
+        return false
+      }
+      return true
     })
-    /* eslint-enable security/detect-object-injection */
+
+    // Only update the file if there were changes
+    if (previousLength !== this.bannedIps.length) {
+      if (verbose) console.log('New banned IPs', this.bannedIps)
+      try {
+        const currentData = this.bannedIps.map((record) => record.ip)
+        fs.writeFileSync('blacklist.json', JSON.stringify(currentData))
+        if (verbose) console.log('Updated blacklist.json with current banned IPs')
+
+        // Verify the bannedIps list is updated
+        const verifyData = JSON.parse(fs.readFileSync('blacklist.json', 'utf8'))
+        if (verifyData.length !== this.bannedIps.length) {
+          if (verbose) console.log('Warning: Inconsistency detected between bannedIps and blacklist.json')
+          // Ensure bannedIps matches the file
+          this.bannedIps = verifyData.map((ip: string) => ({
+            ip,
+            timestamp: Date.now(),
+          }))
```

Comment:
why is this block needed? if this is to handle race condition, then this verification doesn't solve the issue. because `verifyData.length !== this.bannedIps.length` can be true but content of both arrays can be different.